buildstock_fetch/utils/ev_utils.py

```diff
@@ -0,0 +1,297 @@
+"""
+Utility functions for EV demand calculations.
+
+This module contains:
+- Census division mapping functions
+- Data loading functions for metadata, NHTS, and weather data
+"""
+
+from pathlib import Path
+from typing import Any
+
+import polars as pl
+
+__all__ = [
+    "get_census_division_for_state",
+    "load_all_input_data",
+    "load_metadata",
+    "load_metro_puma_map",
+    "load_nhts_data",
+    "load_pums_data",
+    "load_weather_data",
+]
+
+
+def get_census_division_for_state(state: str) -> int:
+    """
+    Get the census division number for a given state.
+
+    Args:
+        state: State abbreviation (e.g., "NY", "CA")
+
+    Returns:
+        Census division number (1-9)
+
+    Raises:
+        ValueError: If the state is not found in any census division
+    """
+    census_divisions = {
+        1: ["CT", "ME", "MA", "NH", "RI", "VT"],  # New England
+        2: ["NJ", "NY", "PA"],  # Middle Atlantic
+        3: ["IL", "IN", "MI", "OH", "WI"],  # East North Central
+        4: ["IA", "KS", "MN", "MO", "NE", "ND", "SD"],  # West North Central
+        5: ["DE", "DC", "FL", "GA", "MD", "NC", "SC", "VA", "WV"],  # South Atlantic
+        6: ["AL", "KY", "MS", "TN"],  # East South Central
+        7: ["AR", "LA", "OK", "TX"],  # West South Central
+        8: ["AZ", "CO", "ID", "MT", "NV", "NM", "UT", "WY"],  # Mountain
+        9: ["AK", "CA", "HI", "OR", "WA"],  # Pacific
+    }
+
+    for division, states in census_divisions.items():
```

Comment:
This whole function should be dictionary that maps state abbreviations to census division numbers. Symbol tables were invented to avoid these kinds of `for` loops! We're trying to have a civilization here. ;)